src/cache/mod.rs

```diff
@@ -122,6 +124,10 @@ pub struct Cache {
     /// received and processed by the cache, the relevant channels are also
     /// removed from this map.
     pub(crate) channels: RwLock<HashMap<ChannelId, GuildChannel>>,
+    /// Cache of channels that have been fetched via to_channel.
+    ///
+    /// Each value has a maximum TTL of 1 hour.
+    pub(crate) temp_channels: RwLock<DashCache<ChannelId, GuildChannel>>,
```

Comment:
```suggestion
    pub(crate) temp_channels: DashCache<ChannelId, GuildChannel>,
```

You do not need to wrap `DashCache` with `RwLock` because `DashCache` is thread-safe and provides interior mutability. After making this change, you will also need to change all `temp_channels.read().await.something()` and `temp_channels.write().await.something()` to `temp_channels.something()` to fix compile errors.

The same things apply to `temp_users`: remove `RwLock` and change `temp_users.read().await` etc.

Reply:
[`76b983e` (#1744)](https://github.com/serenity-rs/serenity/pull/1744/commits/76b983edf4ca8aa5b7ebb847f75fac2e44827868)